ios/RNCAsyncStorage.m

```diff
@@ -343,16 +343,33 @@ - (NSDictionary *)_ensureSetup
     }
     RCTHasCreatedStorageDirectory = YES;
   }
+
   if (!_haveSetup) {
-    NSDictionary *errorOut;
+    NSDictionary *errorOut = nil;
     NSString *serialized = RCTReadFile(RCTGetManifestFilePath(), RCTManifestFileName, &errorOut);
-    _manifest = serialized ? RCTJSONParseMutable(serialized, &error) : [NSMutableDictionary new];
-    if (error) {
-      RCTLogWarn(@"Failed to parse manifest - creating new one.\n\n%@", error);
-      _manifest = [NSMutableDictionary new];
+    if (!serialized) {
+      if (errorOut) {
+        // We cannot simply create a new manifest in case the file does exist but we have no access to it.
```

Comment:
Can we be confident that this is always the case if we get an error? Can we check whether the file exists first?

Reply:
This is what RCTReadFile() is doing — errorOut is nil in case the file does not exist.

Reply:
IMO, it is safer to do an existence check before we read the file rather than relying on the behavior of a private API to stay consistent.

WDYT, @Krizzu?

Reply:
`RCTReadFile()` is our method defined on line 59 above, we can rely on its behavior.

Reply:
Ah, you're absolutely right. I'm blind!